sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2502,3 +2373,140 @@ object UpdateOuterReferences extends Rule[LogicalPlan] {
     }
   }
 }
+
+/**
+ * A helper object used to deduplicate conflicting attributes in logical plans.
+ */
+object DedupQueryAttributesInPlans extends Logging {
```

Comment:
Shall we just move this into a separate file? The analyzer is way to big as it is.

Reply:
Agreed.